iroh-bytes/src/get.rs

```diff
@@ -83,8 +84,12 @@ pub mod fsm {
     }
 
     /// The entry point of the get response machine
-    pub fn start(connection: quinn::Connection, request: GetRequest) -> AtInitial {
-        AtInitial::new(connection, request)
+    pub fn start(
+        connection: quinn::Connection,
+        request: GetRequest,
+        auth: Authenticator,
```

Comment:
If you want to be fancy you could request `impl DynAuthenticator` and then implement it for `Option<Authenticator>` or maybe for `Option<dyn DynAuthenticator>` (not entirely sure what works and what doesn't without trying it out).  It would give a more natural API here as a user would probably guess to pass `None` instead of having to find `NoAuthenticator` and importing it.

Reply:
this doesn't quite work unfortunately, tried it